File: 110-balanced-binary-tree/110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def dfs(root):
            if root is None: return 0
        
            lh = dfs(root.left)
            if lh==-1: return -1
            rh = dfs(root.right)
            if rh==-1: return -1


            if(abs(lh-rh)>1):
                return -1

            return max(lh, rh)+1
        return dfs(root)!=-1
        
        

File: 110-balanced-binary-tree/110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    const rec = (root) => {
        if(root===null){
            return 0
        }

        let lh = rec(root.left)
        if(lh===-1) return -1

        let rh = rec(root.right)    
        if(rh===-1) return -1

        if(Math.abs(lh-rh)>1) return -1

        return 1 + Math.max(lh, rh)
    }
    
    return rec(root)!==-1
};
